### ISSUE TITLE
katharsis-meta improvements

### ISSUE THREAD

**Issue by remmeier** [katharsis-project/katharsis-framework#371]
katharsis-meta currently contains various smaller issues and gaps to be addressed in this ticket (creating 10 tickets out of those seems not appropriate). 

General goal is to complete katharsis-meta and allow JSON API access to all the details of the repositories and resources. Which in turn allows various things: swagger like API documentation, katharsis-ui, API for the generation of client stubs in various languages, automation of UIs (generic tables, editors, etc.). 

The current gaps:
- MetaAttribute should carry indicators whether a field is sortable/filterable/updateable/insertable. For some resources (like the ones derived from JPA entities), this is already clear based on the JPA annotations. For others there will be a need for a @JsonApiField annotation to make that configurable. This mechanism can also be used to perform further validation when parsing URLs.
- non-resource super types of resource should be distinguishable from regular json objects => MetaResourceBase

- resource classses should be plain pojos. All the logic needs to be moved into the MetaProvider implementations.
- primitive types should use camel case for ids, like base.offsetDateTime and base.integer instead of base.offsetdatetime.
- blobs/clobs attributes should be recognizable in the meta model
- validation constraints need to be modelled (notNull, etc.)
- (extensive) documentation explaining the entire model.
- JAXRS and other kind of "action" methods are already available. There parameters not yet. This needs to be properly modelled as well.

ones those tasks are completed, it should be properly reviewed and the experimental flag be removed.
Work will be done on a feature branch till everything is ready (it is being work on in parallel to some so far non-OSS ui automation).

**Comment by masterspambot** [comment]
Please @remmeier explain yourself a little more here. Split the bullets in separate issues and implement them separately so we understand your thinking.

**Comment by remmeier** [comment]
more information added, added documentation will be part of this ticket.

**Comment by remmeier** [comment]
no longer needed
